Updated dependencies ...

mwstowe opened the pull request:
This fit my needs perfectly and functioned as I wanted to, but threw security warnings due to outdated dependencies.

I did the minimum work necessary to bring the dependencies up-to-date, which turned out to be significant.

Have published updated crate as rar-rs, but open to any structure you prefer.  Since it does what I want, I don't plan any further enhancements but will maintain the branch

mwstowe committed:
v0.4.0: Fix CBC decryption and code cleanup

- Fix CBC decryption using decrypt_blocks_mut for proper cipher chain state
- All 32 tests now passing including password-protected archive test
- Bump version to 0.4.0
- Major clippy cleanup: 37% reduction in warnings (19→12)
- Replace manual Default impls with derive macros
- Fix enum naming conventions (WINDOWS→Windows, etc.)
- Remove unused code and fix needless borrows
- Improve Rust idioms and code consistency

mwstowe committed:
Update README for v0.4.0

- Note functional equivalence to v0.3.0
- Highlight updated dependencies and CBC decryption fix
- Mention improved code quality and all tests passing

mwstowe committed:
Update dependencies to latest compatible versions

- lazy_static: 1.1.0 → 1.5.0
- chrono: 0.4.5 → 0.4.42
- All tests still pass (32/32)
- Avoided breaking changes in nom 8.0 and generic-array 1.x

mwstowe committed:
Upgrade to generic-array 1.x

- Update generic-array from 0.14 to 1.3
- Use crypto library's GenericArray for type compatibility
- All 32 tests still pass
- Eliminates user-facing deprecation warnings

mwstowe committed:
Fix clippy unused import warnings

- Use conditional compilation for test-only imports
- Move test helper functions inside test module
- Eliminate 7 unused import warnings
- All 32 tests still pass

mwstowe committed:
Update package metadata

- Fix package name to rar-rs
- Update author email
- Simplify description
- Add missing newline at end of file

mwstowe committed:
Update author name in Cargo.toml

Roba1993 commented on Cargo.toml:
```diff
@@ -1,22 +1,27 @@
-rust-crypto = "0.2.36"
\ No newline at end of file
+nom = "7.1"
+anyhow = "1.0"
```
Because this is a library, anyhow should be replaced with https://docs.rs/thiserror/latest/thiserror/

mwstowe replied:
Ah, excellent point.  Done

Roba1993 commented on Cargo.toml:
```diff
@@ -1,22 +1,27 @@
 readme = "README.md"
-repository = "https://github.com/Roba1993/RAR"
-homepage = "https://github.com/Roba1993/RAR"
+repository = "https://github.com/mwstowe/RAR"
```
For this specific PR, this should still link the right repo

mwstowe replied:
Excellent.  Was hedging my bets a bit in case, after 7 years, you'd abandoned the effort.  Updated.

Roba1993 replied:
Yeah, this is definitely a valid concern 😄

Roba1993 commented on src/lib.rs:
```diff
@@ -24,7 +26,7 @@ mod vint;
 const BUFFER_SIZE: usize = 8192;
 
-use failure::Error;
+use anyhow::Error;
```
Here anyhow should be replaced with this error

mwstowe replied:
Done.

Roba1993 reviewed (commented):
Hi @mwstowe,

thank you so much for the PR. Yes the library is really outdated and desperately needed the update of the dependencies.
To merge it, the cargo.toml for this specific PR for this crate, should also link still to this crate. Also anyhow should be replaced with thiserror, to follow the rust best practices.

Are you willing to make the changes, or should I take over from here?

mwstowe committed:
Replace anyhow with thiserror for better error handling

- Add custom RarError enum with proper error variants
- Replace anyhow::Error with structured error types
- Maintain std::io::Result for trait implementations
- All 32 tests still pass
- Better error messages and type safety

mwstowe committed:
Restore original package name and repository URLs

- Change package name back to 'rar' from 'rar-rs'
- Update repository and homepage to https://github.com/Roba1993/RAR
- All tests still pass

mwstowe committed:
Apply cargo fmt and clippy improvements

- Fix while_let_loop: Convert loop/match to while let pattern
- Fix bool_assert_comparison: Use assert!() instead of assert_eq!(x, true)
- Fix needless_borrows_for_generic_args: Remove unnecessary borrows
- Add #[allow(deprecated)] for GenericArray warnings (crypto compatibility)
- All 32 tests still pass
- Code is now clippy-clean with only expected deprecation warnings

Roba1993 commented:
I published the version 0.4 on crates.io and you should be able to use it easily.

mwstowe commented:
Excellent, thanks!
